stdlib/email/src/main/ballerina/src/email/email_errors.bal

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Record type to hold the details of an error.
+#
+# + message - Specific error message of the error.
+# + cause - Any other error, which causes this error.
+public type Detail record {
+    string message;
+    error cause?;
+};
+
+# SMTP email send error.
+public const EMAIL_SEND_ERROR = "{ballerina/email}EmailSendError";
+# Represents an error that occurs when sending an email fails.
+public type EmailSendError error<EMAIL_SEND_ERROR, Detail>;
+
+# Identifies an email server store access failure error.
+public const GET_STORE_ERROR = "{ballerina/email}GetStoreError";
```

Comment:
Do we have a concept of a store in our email connector?

Reply:
Changed to a different name.